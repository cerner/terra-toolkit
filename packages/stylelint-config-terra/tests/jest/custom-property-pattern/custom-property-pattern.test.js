/**
 * @jest-environment node
 */
const stylelint = require('stylelint');
const path = require('path');

const config = {
  plugins: [
    path.resolve(__dirname, '..', '..', '..', 'src', 'rules', 'custom-property-pattern', 'custom-property-pattern'),
  ],
  rules: {
    'terra/custom-property-pattern': true,
  },
};

describe('custom-property-pattern', () => {
  it('does not error with a custom property', () => {
    const result = stylelint.lint({
      code: '.test { color: var(--test-example-color) }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does not error with another custom property', () => {
    const result = stylelint.lint({
      code: '.test { background-color: var(--test-background-color) }',
      config,
    });

    return result.then(data => expect(data.errored).toBeFalsy());
  });

  it('does error with custom property with underscores', () => {
    const result = stylelint.lint({
      code: '.test { background-color: var(--test_color) }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });

  it('does error with custom property with uppercase characters', () => {
    const result = stylelint.lint({
      code: '.test { color: var(--Test-color) }',
      config,
    });

    return result.then(data => {
      expect(data.errored).toBeTruthy();
      expect(data.results[0].warnings).toMatchSnapshot();
    });
  });
});
