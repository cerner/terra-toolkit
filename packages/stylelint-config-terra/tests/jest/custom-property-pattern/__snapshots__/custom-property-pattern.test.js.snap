// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-property-pattern does error with custom property with underscores 1`] = `
Array [
  Object {
    "column": 9,
    "line": 1,
    "rule": "terra/custom-property-pattern",
    "severity": "error",
    "text": "Expected --test_color to include only lowercase alphanumeric characters and hyphens (terra/custom-property-pattern)",
  },
]
`;

exports[`custom-property-pattern does error with custom property with uppercase characters 1`] = `
Array [
  Object {
    "column": 9,
    "line": 1,
    "rule": "terra/custom-property-pattern",
    "severity": "error",
    "text": "Expected --Test-color to include only lowercase alphanumeric characters and hyphens (terra/custom-property-pattern)",
  },
]
`;
